### PR TITLE
Updates in gitlab config for SDK CI @LLNL

### PR DESCRIPTION
**Pull request by mtitov**
~@jameshcorbett James, can you please also reset file permissions with `chmod -fR 02770 .` in `/usr/workspace/exaworks/sdk/spack`~

### CODE REVIEW

**Comment by jameshcorbett** [comment]
Done

**Comment by mtitov** [comment]
reworked:
- all processes are automated (setup processes, which were required to be executed manually, are moved into the script);
- updated spack env setup and installation;
- ensured correctness of FS permissions;

ready for review @jameshcorbett @stephdempsey

File: .gitlab/llnl-ci.yml [hunk 1]
```diff
@@ -1,79 +1,132 @@
...
   tags:
     - quartz
     - shell
   script:
-    - cd $SPACK_WORK_DIR/spack/
-    - git checkout HEAD^ .
-    - git pull --ff
-    - chmod -fR 02770 . || true
+    - mkdir -p $SPACK_WORK_DIR
+    - test -d $SPACK_WORK_DIR/spack && exit 0
```
Comment (jameshcorbett) [inline .gitlab/llnl-ci.yml:21]: If the directory exists, it's assumed to be the repo, so quit?

Just making sure I understand. I assume the sequence of commands here does not abort on the first error?

File: .gitlab/llnl-ci.yml [hunk 2]
```diff
@@ -1,79 +1,132 @@
...
+        LLNL_SLURM_SCHEDULER_PARAMETERS: [ "--nodes=1 -t 30" ]
   tags:
-    - batch
     - $HOST
+    - batch
   script:
     - . $SPACK_WORK_DIR/spack/share/spack/setup-env.sh
-    - spack env activate $SPACK_ENV_NAME
+    - for i in {1..3}; do
+        spack env activate $SPACK_ENV_NAME && break || sleep 5;
```
Comment (jameshcorbett) [inline .gitlab/llnl-ci.yml:128]: What is the point of this loop?

**Review by jameshcorbett — APPROVED** [review]
Looks reasonable to me, two little questions below.

File: .gitlab/llnl-ci.yml (hunk 1 above)
Comment (mtitov) [reply .gitlab/llnl-ci.yml:21]: correct, this stage is responsible to clone the spack repo, automates the cloning process if need to run the pipeline after removing the spack dir

File: .gitlab/llnl-ci.yml (hunk 2 above)
Comment (mtitov) [reply .gitlab/llnl-ci.yml:128]: we have 3 parallel jobs here and each of them uses the same spack setup, it happened they were interfering with each other, thus here 3 attempts to activate a corresponding spack env

**Pull request merged**